index.bs

```diff
@@ -1194,6 +1194,13 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
                         :   {{AuthenticatorAttestationResponse/attestationObject}}
                         ::  |attestationObject|
 
+                        :   {{AuthenticatorAttestationResponse/transports}}
+                        ::  A sequence of one or more {{AuthenticatorTransport}} values. The first value is set to the transport used to obtain this registration. Further values are appended to indicate other, distinct, transports that the |authenticator| is believed to support.
```

Comment:
>The first value is set to the transport used to obtain this registration. Further values are appended to indicate other, distinct, transports [...]

I think we can leverage the types better to express this. I think that as a rule of thumb, particular items in a sequence should not be special because that speciality is not encoded in the type. I therefore suggest removing the significance of order in the list and instead adding a separate property that indicates the transport that was used this time. Something like:

```
{
  "transports": ["USB", "NFC"],
  "usedTransport": "USB"
}
```

This would also let us solve the issue of possible fingerprintability based on order, since we could then just specify that the list MUST be sorted lexicographically for example. The `usedTransport` member could also be optional instead of letting the client "choose the first element to be a generic value rather than the actual transport used".

Then again, the first element is special also in the `x5c` member of some attestation statement formats, so it's not a huge deal if we leave this as is.

Reply:
In summary my position is that special cases inevitably cause bugs, so we should stay away from special cases whenever possible.

Reply:
I disagree with this :) At the moment, there are two ways in which `transports` can be invalid: being empty or having duplicates. If the transport that was used is indicated by setting it as the value of a different member then there's a third possible problem: that the used transport isn't included in the list of transports.

Also, when this is exposed as API it has to be as a `getTransports` function because we can't have a sequence as a property of an interface. Since this isn't a feature of the baseline protocol, RP Javascript will have to test for `"getTransports" in response`, but now there'll be two properties to worry about. Or maybe return [usedTransport, transports] from `getTransports`? (But tuple-returns don't seem very common in Javascript.)

This is all pretty minor, but I do think that assigning semantics to the order of elements in a list is pretty standard and it's simpler to have a single list.

Reply:
I prefer emlun approach. This is much cleaner. @agl, I still don't understand the "Generic Value" part of the PR. Can you explain what that value is as we cannot use "usb" as Generic Value.

Reply:
@agl There would be equally many (2) invalid configurations, because the empty list would be valid. Even if this means there's two properties to worry about, I argue that it _reduces_ complexity because it _increases_ explicitness. And I'm not convinced that just because something is common practice that means it's good practice. :)

Reply:
The empty list would not be valid based on the above. However, I think we can solve this by dropping the used transport completely. I had included it because it seemed that it might be useful, but we don't have a concrete use. Without it, the list is just a sorted list of transports. I think everyone will be happy with that and I'll upload a version with that in a sec.

I have made the empty list valid for a different reason however: I suspect that this information may not be valid in some cases. For example, platform APIs written against the first version of webauthn may not include this information and so the assumption that at least the transport used to register a credential is known, might not hold. Thus I've defined the empty list to mean handle that case.